lib/src/util/posix/pid_file.cc

```diff
@@ -27,21 +27,6 @@ PIDFile::PIDFile(const std::string& file_path_)
           file_path { file_path_ },
           pidfile_fd {},
           cleanup_when_done { false } {
-    if (fs::exists(file_path) && !(fs::is_regular_file(file_path))) {
```

Comment:
Moved to the Configuration::validate() method, since we don't check this in the unit tests anymore.
It is better to break sooner in case of invalid PID dir.